services/app-api/src/dataMigraitons/dataMigrator.ts

```diff
@@ -0,0 +1,140 @@
+import fs from 'fs'
+
+import { PrismaClient } from '@prisma/client'
+import type { PrismaTransactionType } from '../postgres/prismaTypes'
+
+// MigrationType describes a single migration with a name and a callable function called migrateProto
+interface DBMigrationType {
+    name: string
+    module: {
+        migrate: (client: PrismaTransactionType) => Promise<undefined | Error>
+    }
+}
+
+// MigratorType is a type covering our two different migrators
+export interface MigratorType {
+    listMigrationsThatHaveRun(): Promise<string[]>
+    runMigrations(migrations: DBMigrationType[]): Promise<undefined | Error>
+}
+
+export function newDBMigrator(dbConnString: string): MigratorType {
+    const prismaClient = new PrismaClient({
+        datasources: {
+            db: {
+                url: dbConnString,
+            },
+        },
+    })
+
+    return {
+        async listMigrationsThatHaveRun(): Promise<string[]> {
+            const listMigrationsThatHaveRunTable =
+                await prismaClient.protoMigrationsTable.findMany()
+            const migrations = listMigrationsThatHaveRunTable.map(
+                (m) => m.migrationName
+            )
+
+            return migrations
+        },
+
+        async runMigrations(migrations: DBMigrationType[]) {
+            for (const migration of migrations) {
+                try {
+                    await prismaClient.$transaction(async (tx) => {
+                        const res = await migration.module.migrate(tx)
+
+                        if (!(res instanceof Error)) {
+                            await tx.protoMigrationsTable.create({
+                                data: {
+                                    migrationName: migration.name,
+                                },
+                            })
+                        } else {
+                            console.error('migrator error:', res)
```

Comment:
Is the name of the migration that failed  in the output here? I feel like that would help with debugging.